### PR TITLE
Fix #1910 - Review and improve docs for 'Order' properties

### LINKED ISSUE #1910: Review use of Order
Generated through discussion:

We should review the the use of `Order` properties where they appear in the framework for consistency. 

Examples:
- Attribute Routing
- Nested Providers
  - Action Descriptor Provider
  - API Description Provider
  - Action Invoker Provider
  - Filter Provider
  - Action Constraint Provider
- Filters
- Options
- Action Constraints

### PR DESCRIPTION
**Pull request by rynowak**

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Abstractions/IActionConstraint.cs [hunk 1]
```diff
@@ -39,6 +39,10 @@ public interface IActionConstraint : IActionConstraintMetadata
         /// <summary>
         /// The constraint order.
         /// </summary>
+        /// <remarks>
+        /// Constraints are grouped into stages by the value of <see cref="Order"/>. See remarks on
+        /// <see cref="IActionConstraint"/>.
+        /// </remarks>
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Abstractions/IActionConstraint.cs]: This already has pretty comprehensive docs.

File: src/Microsoft.AspNet.Mvc.Abstractions/IActionConstraintProvider.cs [hunk 2]
```diff
@@ -7,7 +7,27 @@ namespace Microsoft.AspNet.Mvc.ActionConstraints
...
+        /// all calls to <see cref="OnProvidersExecuting"/>. A provider with a lower numeric value of
+        /// <see cref="Order"/> will have its <see cref="OnProvidersExecuted"/> method called after that of a provider
+        /// with a higher numeric value of <see cref="Order"/>.
+        /// </para>
+        /// <para>
+        /// If two providers have the same numeric value of <see cref="Order"/>, then their relative execution order
+        /// is undefined.
+        /// </para>
+        /// </remarks>
         int Order { get; }
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Abstractions/IActionConstraintProvider.cs]: All of the former nested providers are copy-pasta

**Comment by Eilon** [comment]
Very nice.

File: src/Microsoft.AspNet.Mvc.Core/IActionDescriptorProvider.cs [hunk 3]
```diff
@@ -7,6 +7,25 @@ namespace Microsoft.AspNet.Mvc.Core
...
+        /// <see cref="Order"/> will have its <see cref="OnProvidersExecuted"/> method called after that of a provider
+        /// with a higher numeric value of <see cref="Order"/>.
+        /// </para>
+        /// <para>
+        /// If two providers have the same numeric value of <see cref="Order"/>, then their relative execution order
+        /// is undefined.
+        /// </para>
+        /// </remarks>
         int Order { get; }
         void OnProvidersExecuting([NotNull] ActionDescriptorProviderContext context);
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Core/IActionDescriptorProvider.cs]: that blank line you added in other similar classes was a good idea

**Comment by dougbu** [comment]
:shipit:

**Pull request closed without merging**